sdk/python/src/dstack_sdk/dstack_client.py

```diff
@@ -114,118 +181,100 @@ class InfoResponse(BaseModel):
     compose_hash: str
 
     @classmethod
-    def model_validate(cls, obj: Any) -> 'InfoResponse':
-        if isinstance(obj, dict) and 'tcb_info' in obj and isinstance(obj['tcb_info'], str):
+    def parse_response(cls, obj: Any) -> "InfoResponse":
+        """Parse raw JSON into ``InfoResponse``, decoding nested ``tcb_info``."""
+        if (
+            isinstance(obj, dict)
+            and "tcb_info" in obj
+            and isinstance(obj["tcb_info"], str)
+        ):
             obj = dict(obj)
-            obj['tcb_info'] = TcbInfo(**json.loads(obj['tcb_info']))
-        return super().model_validate(obj)
+            obj["tcb_info"] = TcbInfo(**json.loads(obj["tcb_info"]))
+        return cls(**obj)
 
 
 class BaseClient:
-    pass
+    """Marker base class for dstack clients."""
 
 
 class DstackClient(BaseClient):
+    """Synchronous client for dstack services."""
+
     def __init__(self, endpoint: str | None = None):
+        """Initialize client with HTTP or Unix-socket transport.
+
+        If a non-HTTP(S) endpoint is provided, it is treated as a Unix socket
+        path and validated for existence.
+        """
         endpoint = get_endpoint(endpoint)
-        if endpoint.startswith("http://") or endpoint.startswith('https://'):
+        if endpoint.startswith("http://") or endpoint.startswith("https://"):
             self.transport = httpx.HTTPTransport()
             self.base_url = endpoint
         else:
+            # Check if Unix socket file exists
+            if endpoint.startswith("/") and not os.path.exists(endpoint):
+                raise FileNotFoundError(f"Unix socket file {endpoint} does not exist")
             self.transport = httpx.HTTPTransport(uds=endpoint)
             self.base_url = "http://localhost"
 
-    def _send_rpc_request(self, path, payload):
+    def _send_rpc_request(self, path: str, payload: Dict[str, Any]) -> Dict[str, Any]:
+        """Send an RPC request and return the parsed JSON response."""
```

Comment:
```suggestion
    def _send_rpc_request(self, method: str, payload: Dict[str, Any]) -> Dict[str, Any]:
        """Send an RPC request and return the parsed JSON response."""
        path = self.PATH_PREFIX + method
```

And all the caller site need to update.